pkg/client/client.go

```diff
@@ -72,6 +78,27 @@ func (i InitialiseClient) NewClient(r *schema.ResourceData) (interface{}, error)
 	utils.SetMeta(apiClient, r)
 	client.CmpClient = cmp_client.NewClient(apiClient, cfg)
 
+	// Create broker client
+	brokerHeaders := getHeaders()
+	tenantID := r.Get(constants.TenantID).(string)
+	brokerHeaders["X-Tenant-ID"] = tenantID
+	// We don't add default query params to broker client
+	brokerCfgForAPIClient := api_client.Configuration{
+		Host:          vmaasProviderSettings[constants.BROKERRURL].(string),
+		DefaultHeader: brokerHeaders,
+	}
+	brokerApiClient := api_client.NewAPIClient(&brokerCfgForAPIClient)
+	utils.SetMetaFnAndVersion(brokerApiClient, r, apiClient.GetSCMVersion())
+
+	// brokerCfg is passed down to NewBrokerClient.  It is used mainly to set the query params
+	// for the call to get SubscriptionDetails
+	brokerCfg := api_client.Configuration{
+		Host:               vmaasProviderSettings[constants.BROKERRURL].(string),
```

Comment:
Please introduce new field at the vmaas provider level so that we can configure this broker url in case of dpce as well with env vars.